scripts/exregional_run_met_ascii2nc_obs.sh

```diff
@@ -0,0 +1,376 @@
+#
+# Make sure the MET/METplus output directory(ies) exists.
+#
+#-----------------------------------------------------------------------
+#
+mkdir -p "${OUTPUT_DIR}"
+#
+#-----------------------------------------------------------------------
+#
+# Check for existence of top-level OBS_DIR.
+#
+#-----------------------------------------------------------------------
+#
+if [ ! -d "${OBS_DIR}" ]; then
+  print_err_msg_exit "\
+OBS_DIR does not exist or is not a directory:
+  OBS_DIR = \"${OBS_DIR}\""
+fi
+#
+#-----------------------------------------------------------------------
+#
+# Export variables needed in the common METplus configuration file (at
+# ${METPLUS_CONF}/common.conf).
+#
+#-----------------------------------------------------------------------
+#
+export METPLUS_CONF
+export LOGDIR
+#
+#-----------------------------------------------------------------------
+#
+# Do not run METplus if there isn't at least one valid forecast hour for
+# which to run it.
+#
+#-----------------------------------------------------------------------
+#
+if [ -z "${LEADHR_LIST}" ]; then
+  print_err_msg_exit "\
+The list of forecast hours for which to run METplus is empty:
+  LEADHR_LIST = [${LEADHR_LIST}]"
+fi
+#
+#-----------------------------------------------------------------------
+#
+# Set the names of the template METplus configuration file, the METplus
+# configuration file generated from this template, and the METplus log
+# file.
+#
+#-----------------------------------------------------------------------
+#
+# First, set the base file names.
+#
+metplus_config_tmpl_fn="${MetplusToolName}_obs"
+#
+# Set the name of the final conf file that will be used for this task. We
+# append the OBTYPE and cycle date to ensure that different tasks in the same
+# workflow won't overwrite each others' conf files.
+#
+metplus_config_fn="${metplus_config_tmpl_fn}_${OBTYPE}_${CDATE}"
+metplus_log_fn="${metplus_config_fn}"
+#
+# Add prefixes and suffixes (extensions) to the base file names.
+#
+metplus_config_tmpl_fn="${metplus_config_tmpl_fn}.conf"
+metplus_config_fn="${metplus_config_fn}.conf"
+metplus_log_fn="metplus.log.${metplus_log_fn}"
+#
+#-----------------------------------------------------------------------
+#
+# Generate the METplus configuration file from its jinja template.
+#
+#-----------------------------------------------------------------------
+#
+# Set the full paths to the jinja template METplus configuration file
+# (which already exists) and the METplus configuration file that will be
+# generated from it.
+#
+metplus_config_tmpl_fp="${METPLUS_CONF}/${metplus_config_tmpl_fn}"
+metplus_config_fp="${OUTPUT_DIR}/${metplus_config_fn}"
+#
+# Define variables that appear in the jinja template.
+#
+settings="\
+#
+# MET/METplus information.
+#
+  'metplus_tool_name': '${metplus_tool_name}'
+  'MetplusToolName': '${MetplusToolName}'
+  'METPLUS_TOOL_NAME': '${METPLUS_TOOL_NAME}'
```

Comment:
Why are all three of these needed? Seems like the same information repeated a lot.

Reply:
Unfortunately MET and METplus have a mix of capitalization and punctuation conventions for tools at various layers, so we need to maintain separate entries here. There may be more elegant ways to handle this, but this actually prevents the repetition of a lot of logic.